pkg/sql/show_tenant.go

```diff
@@ -124,89 +152,103 @@ func (n *showTenantNode) initReplicationStats(params runParams, job *jobs.Job) e
+	var values tenantValues
+	values.tenantInfo = tenantRecord
+	jobId := values.tenantInfo.TenantReplicationJobID
 	if jobId == 0 {
 		// No replication job, this is a non-replicating tenant.
 		if n.withReplication {
-			return errors.Newf("tenant %q does not have an active replication job", tenantName)
+			return nil, errors.Newf("tenant %q is not a replicated tenant", tenantName)
 		}
-		n.tenantStatus = tenantStatus(n.tenantInfo.State.String())
-		return nil
+		values.tenantStatus = tenantStatus(values.tenantInfo.State.String())
+		return &values, nil
 	}
 
-	switch n.tenantInfo.State {
+	switch values.tenantInfo.State {
 	case descpb.TenantInfo_ADD:
 		// There is a replication job, we need to get the job info and the
 		// replication stats in order to generate the exact tenant status.
 		registry := params.p.execCfg.JobRegistry
 		job, err := registry.LoadJobWithTxn(params.ctx, jobId, params.p.Txn())
 		if err != nil {
-			return err
+			values.tenantStatus = tenantStatus(fmt.Sprintf(string(replicationUnknownFormat), err))
+			return &values, nil
 		}
-		if err := n.initReplicationStats(params, job); err != nil {
-			return err
+		stats, protectedTimestamp, err := getReplicationStats(params, job)
+		if err != nil {
+			values.tenantStatus = tenantStatus(fmt.Sprintf(string(replicationUnknownFormat), err))
+			return &values, nil
+		}
+		values.replicationInfo = stats
+		if protectedTimestamp != nil {
+			values.protectedTimestamp = *protectedTimestamp
 		}
 
 		switch job.Status() {
 		case jobs.StatusPending, jobs.StatusRunning, jobs.StatusPauseRequested:
-			if n.replicationInfo == nil || n.replicationInfo.ReplicationLagInfo == nil {
+			if values.replicationInfo == nil || values.replicationInfo.ReplicationLagInfo == nil {
 				// Still no lag info which means we never recorded progress, and
 				// replication did not complete the initial scan yet.
-				n.tenantStatus = initReplication
+				values.tenantStatus = initReplication
 			} else {
-				progress := n.replicationInfo.IngestionProgress
+				progress := values.replicationInfo.IngestionProgress
 				if progress != nil && !progress.CutoverTime.IsEmpty() {
-					n.tenantStatus = cuttingOver
+					values.tenantStatus = cuttingOver
 				} else {
-					n.tenantStatus = replicating
+					values.tenantStatus = replicating
 				}
 			}
 		case jobs.StatusPaused:
-			n.tenantStatus = replicationPaused
+			values.tenantStatus = replicationPaused
 		default:
-			n.tenantStatus = tenantStatus(fmt.Sprintf(string(replicationUnknownFormat), job.Status()))
+			values.tenantStatus = tenantStatus(fmt.Sprintf(string(replicationUnknownFormat), job.Status()))
 		}
 	case descpb.TenantInfo_ACTIVE, descpb.TenantInfo_DROP:
-		n.tenantStatus = tenantStatus(n.tenantInfo.State.String())
+		values.tenantStatus = tenantStatus(values.tenantInfo.State.String())
 	default:
-		return errors.Newf("unknown tenant status %s", n.tenantInfo.State)
+		status := fmt.Sprintf("tenant status: %s", values.tenantInfo.State)
```

Comment:
I prefer if we leave this an error. If we see a tenant in a state other than `ADD`, `DROP`, `ACTIVE` then it means someone has added a state and not considered all the places it effects, so I'd rather it fail tests loudly. wdyt?